ev3dev2/motor.py

```diff
@@ -2670,66 +2761,26 @@ def _odometry_monitor():
                 if sleep_time:
                     time.sleep(sleep_time)
 
-            self.odometry_thread_id = None
+        _thread.start_new_thread(_odometry_monitor, ())
 
-        self.odometry_thread_run = True
-        self.odometry_thread_id = _thread.start_new_thread(_odometry_monitor, ())
+        # Block until the thread has started doing work
+        while not self.odometry_thread_run:
+            pass
 
     def odometry_stop(self):
         """
-        Signal the odometry thread to exit and wait for it to exit
+        Signal the odometry thread to exit
         """
 
-        if self.odometry_thread_id:
+        if self.odometry_thread_run:
             self.odometry_thread_run = False
 
-            while self.odometry_thread_id:
-                pass
-
-    def turn_to_angle(self, speed, angle_target_degrees, brake=True, block=True):
-        """
-        Rotate in place to ``angle_target_degrees`` at ``speed``
-        """
-        assert self.odometry_thread_id, "odometry_start() must be called to track robot coordinates"
-
-        # Make both target and current angles positive numbers between 0 and 360
-        if angle_target_degrees < 0:
-            angle_target_degrees += 360
-
-        angle_current_degrees = math.degrees(self.theta)
-
-        if angle_current_degrees < 0:
-            angle_current_degrees += 360
-
-        # Is it shorter to rotate to the right or left
-        # to reach angle_target_degrees?
-        if angle_current_degrees > angle_target_degrees:
-            turn_right = True
-            angle_delta = angle_current_degrees - angle_target_degrees
-        else:
-            turn_right = False
-            angle_delta = angle_target_degrees - angle_current_degrees
-
-        if angle_delta > 180:
-            angle_delta = 360 - angle_delta
-            turn_right = not turn_right
-
-        log.debug("%s: turn_to_angle %s, current angle %s, delta %s, turn_right %s" %
-            (self, angle_target_degrees, angle_current_degrees, angle_delta, turn_right))
-        self.odometry_coordinates_log()
-
-        if turn_right:
-            self.turn_right(speed, angle_delta, brake, block)
-        else:
-            self.turn_left(speed, angle_delta, brake, block)
-
-        self.odometry_coordinates_log()
-
     def on_to_coordinates(self, speed, x_target_mm, y_target_mm, brake=True, block=True):
         """
         Drive to (``x_target_mm``, ``y_target_mm``) coordinates at ``speed``
         """
-        assert self.odometry_thread_id, "odometry_start() must be called to track robot coordinates"
+        if not self.odometry_thread_run:
+            raise ThreadNotRunning("odometry_start() must be called to track robot coordinates")
```

Comment:
This could be breaking but seems unlikely someone would be catching this assert when trying to use odometry.